### PR TITLE
Upgrade parser, unicode strings, catch Panic

### PR DESCRIPTION
**Pull request by fvictorio**
This PR upgrades the parser and adds support for its two new features:

- Caught `Panic` errors are properly printed
- Unicode strings are handled

### CODE REVIEW

**Comment by fvictorio** [comment]
@Janther @mattiaerre can we remove the codecov bot? I think it's more annoying that useful (I don't really mind keeping it though, it's just to avoid the mild inconvinience of receiving a notification after each PR)

**Comment by Janther** [comment]
Happy to merge this one and also happy to remove the `codecov` as long as we still keep the coverage while running tests.

**Review by Janther — APPROVED** [review]

**Comment by fvictorio** [comment]
Yes, I think having a threshold and check that is way better. I added a commit that does that (it also uses `test:all` for the CI, that enables `AST_COMPARE`).

**Comment by fvictorio** [comment]
Yes, we have 100% coverage. Quite amazing.

File: jest.config.js [hunk 1]
```diff
@@ -8,6 +8,14 @@ module.exports = {
...
   ],
   coverageDirectory: './coverage/',
+  coverageThreshold: {
+    global: {
+      branches: 100,
+      functions: 100,
+      lines: 100,
+      statements: 100
+    }
+  },
```
Comment (mattiaerre) [inline jest.config.js:18]: WOW 🎉

**Review by mattiaerre — APPROVED** [review]
this is really cool thanks @fvictorio

File: .travis.yml [hunk 2]
```diff
@@ -7,4 +7,4 @@ node_js:
   - '13'
 script:
   - npm run lint
-  - npm test
+  - npm run test:all
```
[the remark this answers is not in the document]
Comment (fvictorio) [reply .travis.yml:10]: I changed this because that way the coverage threshold will be checked.

**Comment by mattiaerre** [comment]
happy to merge 👍

**Comment by mattiaerre** [comment]
> Yes, we have 100% coverage. Quite amazing.

yeah we had it since https://github.com/prettier-solidity/prettier-plugin-solidity/pull/414

**Pull request merged**